robocluster/device.py

```diff
@@ -111,6 +114,32 @@ def sleep(self, duration):
             for callback in self.events[event]:
                 self._loop.create_task(callback(event, data))
 
+    async def _serial_read_task(self):
+        """
+        Recieve packets from a linked serial device
+        and call the appropriate callbacks.
+        """
+        while not self._serial_device.isInitialized():
+            await self.sleep(0.01)  # Better way to do this?
```

Comment:
[Asynchronous context managers](https://docs.python.org/3.5/reference/datamodel.html#asynchronous-context-managers) may be the better solution here.